### PR TITLE
Use assertions on Vec doc

### PR DESCRIPTION
**Pull request by pickfire**
Clarify what the state of Vec after with_capacity on doc.

r? @jyn514

### CODE REVIEW

File: library/alloc/src/vec.rs [hunk 1]
```diff
@@ -116,8 +116,10 @@ use crate::raw_vec::RawVec;
 /// assert_eq!(vec, [0, 0, 0, 0, 0]);
 ///
 /// // The following is equivalent, but potentially slower:
-/// let mut vec1 = Vec::with_capacity(5);
-/// vec1.resize(5, 0);
+/// let mut vec = Vec::with_capacity(5);
+/// assert_eq!(vec, []);
```
Comment (jyn514) [inline library/alloc/src/vec.rs:120]: I think this assert is a little redundant, `with_capacity` already has an example like this and it interrupts the flow of the example.
```suggestion
```

**Review by jyn514 — COMMENTED** [review]
r=me after nit fixed

File: library/alloc/src/vec.rs (hunk 1 above)
Comment (pickfire) [reply library/alloc/src/vec.rs:120]: I understand but this is just to clarify to the readers. This section appears before `with_capacity` or any discussion related to capacity so we assumed the reader does not know what is `with_capacity`. It does disrupt the flow a bit even I was confused what `resize` does so just to confirm I add this.

Comment (jyn514) [reply library/alloc/src/vec.rs:120]: Hmm, instead of adding the assert could we link to the capacity section? I think that would be more helpful.

Comment (pickfire) [reply library/alloc/src/vec.rs:120]: `#capacity-and-reallocation` do you have any idea where should we put that?

Comment (jyn514) [reply library/alloc/src/vec.rs:120]: Maybe `performing allocation and initialization in separate steps` could be a link? Or just `performing allocation` if that's too long?

Comment (pickfire) [reply library/alloc/src/vec.rs:120]: I will just commit this, maybe you can follow up on that.

**Comment by jyn514** [comment]
@bors r+ rollup

@jyn514 note to self - follow up on https://github.com/rust-lang/rust/pull/76058#discussion_r479655750

**Pull request merged**